src/gpodder/plugins/soundcloud.py

```diff
@@ -142,9 +149,14 @@ def get_tracks(self, feed):
                 logger.debug('Skipping track with no base_url')
                 continue
 
-            logger.debug('track in tracks url: %s', url)
+            track_guid = track.get('permalink', track.get('id'))
 
-            filesize, filetype, filename = get_metadata(url)
+            if track_guid not in existing_guids:
+                filesize, filetype, filename = get_metadata(url)
+            else:
+                filesize = existing_guids[track_guid]['filesize']
+                filetype = existing_guids[track_guid]['filetype']
+                logger.debug('Skipping existing track - %s', url)
```

Comment:
This logging might be quite verbose, as I'd imagine this code path is hit often ideally?

Reply:
Absolutely, I left it in because I figured it would be easier if someone is trying to figure out why they aren't getting an episode if they have a full list of "ignored" episodes.

Reply:
Do you think people would benefit from this verbosity?

Reply:
I don't know.. maybe you could only print one log message at the end instead of for each track, something like "%d tracks read from cache" or so?

Reply:
Done